libs/domain-service/src/utils/index.ts

```diff
@@ -1,3 +1,30 @@
+import { PromiseResult } from "../types";
+
+/**
+ * Needed polyfill for Promise.allSettled as it doesn't exist on RN.
+ */
+export const allSettled = <T>(
+  promises: Promise<T>[]
+): Promise<PromiseResult<T>[]> => {
+  return Promise.all(
```

Comment:
is it possible to use the node native one if it's exist, otherwise use this one ?